builder/test/pipeline.cc

```diff
@@ -578,10 +583,14 @@ TEST(pyramid, pipeline) {
   const raw_buffer* outputs[] = {&out_buf};
   test_context eval_ctx;
   p.evaluate(inputs, outputs, eval_ctx);
-  ASSERT_EQ(eval_ctx.heap.total_size, (W + 2) / 2 * 2 * sizeof(int));
-  ASSERT_EQ(eval_ctx.heap.total_count, 1);
 
-  check_replica_pipeline(define_replica_pipeline(ctx, {in}, {out}));
+  ASSERT_EQ(eval_ctx.heap.total_count, 1);
+  if (max_workers == loop::serial) {
+    ASSERT_EQ(eval_ctx.heap.total_size, (W + 2) / 2 * 2 * sizeof(int));
```

Comment:
Nit: Please write this assert outside the if, and use a condition to choose between 2 or 3, so it's clear the only thing that changes is the fold factor in y. See https://github.com/dsharlet/slinky/blob/21dc9b0c52e7d519e75e05fde11015ed7410d5b4/builder/test/pipeline.cc#L641-L642 for example.